hbase-client/src/main/java/org/apache/hadoop/hbase/CatalogFamilyFormat.java

```diff
@@ -0,0 +1,349 @@
+ * {@value org.apache.hadoop.hbase.HConstants#CATALOG_FAMILY_STR} family cells in meta table.
+ * <p/>
+ * The cells in catalog family are:
+ *
+ * <pre>
+ * For each table range ('Region'), there is a single row, formatted as:
+ * &lt;tableName&gt;,&lt;startKey&gt;,&lt;regionId&gt;,&lt;encodedRegionName&gt;.
+ * This row is the serialized regionName of the default region replica.
+ * Columns are:
+ * info:regioninfo         => contains serialized HRI for the default region replica
+ * info:server             => contains hostname:port (in string form) for the server hosting
+ *                            the default regionInfo replica
+ * info:server_&lt;replicaId&gt => contains hostname:port (in string form) for the server hosting
+ *                                 the regionInfo replica with replicaId
+ * info:serverstartcode    => contains server start code (in binary long form) for the server
+ *                            hosting the default regionInfo replica
+ * info:serverstartcode_&lt;replicaId&gt => contains server start code (in binary long form) for
+ *                                          the server hosting the regionInfo replica with
+ *                                          replicaId
+ * info:seqnumDuringOpen   => contains seqNum (in binary long form) for the region at the time
+ *                            the server opened the region with default replicaId
+ * info:seqnumDuringOpen_&lt;replicaId&gt => contains seqNum (in binary long form) for the region
+ *                                           at the time the server opened the region with
+ *                                           replicaId
+ * info:splitA             => contains a serialized HRI for the first daughter region if the
+ *                            region is split
+ * info:splitB             => contains a serialized HRI for the second daughter region if the
+ *                            region is split
+ * info:merge*             => contains a serialized HRI for a merge parent region. There will be two
+ *                            or more of these columns in a row. A row that has these columns is
+ *                            undergoing a merge and is the result of the merge. Columns listed
+ *                            in marge* columns are the parents of this merged region. Example
+ *                            columns: info:merge0001, info:merge0002. You make also see 'mergeA',
+ *                            and 'mergeB'. This is old form replaced by the new format that allows
+ *                            for more than two parents to be merged at a time.
+ * </pre>
+ */
+@InterfaceAudience.Private
+public class CatalogFamilyFormat {
+
+  private static final Logger LOG = LoggerFactory.getLogger(CatalogFamilyFormat.class);
+
+  /** A regex for parsing server columns from meta. See above javadoc for meta layout */
+  private static final Pattern SERVER_COLUMN_PATTERN =
+    Pattern.compile("^server(_[0-9a-fA-F]{4})?$");
```

Comment:
replica_id is a integer? So it should only have [0-9]?

Reply:
@InterfaceAudience.Private
  String REPLICA_ID_FORMAT = "%04X";

 I think it is in hex format?